python/dgl/dataloading/dataloader.py

```diff
@@ -336,8 +339,41 @@ def __len__(self):
         ) // self.batch_size
 
 
+def _numel_of_shape(shape):
+    return reduce(operator.mul, shape, 1)
+
+
+def _init_gpu_caches(graph, gpu_caches):
+    if not hasattr(graph, "_gpu_caches"):
+        graph._gpu_caches = {}, {}
+    if gpu_caches is None:
+        return
+    if not isinstance(gpu_caches, tuple):
```

Comment:
Is it implicitly indicates when `gpu_cahches` is a single, it must be a node cache?

Reply:
Yes, I assumed that would be the most used case. It is a single dict in this case which looks very clean.

Reply:
An example usage of this feature is in the labor example PR #4718.